backend/routes/reservation_routes.py

```
@@ -274,14 +297,14 @@ def create_reservation():
 
     except ValueError:
         session.rollback()
-        return jsonify({"error": "Invalid date format. Use 'YYYY-MM-DD'"}), 400
+        return error_response("Invalid date format. Use 'YYYY-MM-DD'", 400)
     except KeyError as e:
         session.rollback()
-        return jsonify({"error": f"Missing key: {str(e)}"}), 400
+        return error_response(f"Missing key: {str(e)}", 400)
     # pylint: disable=W0718
     except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)